tls/s2n_client_extensions.c

```diff
@@ -87,10 +88,13 @@ int s2n_client_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *
     const struct s2n_cipher_preferences *cipher_preferences;
     GUARD(s2n_connection_get_cipher_preferences(conn, &cipher_preferences));
 
+    const struct s2n_ecc_preferences *ecc_pref = conn->config->ecc_preferences;
+    notnull_check(ecc_pref);
+
     const uint8_t ecc_extension_required = s2n_ecc_extension_required(cipher_preferences);
     if (ecc_extension_required) {
         /* Write ECC extensions: Supported Curves and Supported Point Formats */
-        total_size += 12 + s2n_ecc_evp_supported_curves_list_len * 2;
+        total_size += 12 + ecc_pref->count * 2;
```

Comment:
Is the 12 support group / keyshare?

Reply:
Good Question, I am not entirely sure what the 12 refers to here.

Reply:
see https://github.com/awslabs/s2n/blob/master/tls/extensions/s2n_client_supported_groups.c#L27-L41

you could write 12 as
```
5 * sizeof(uint16_t) + 2 * sizeof(uint8_t)
```